### PR TITLE
[Telemetry] Use imdsv2 first and then imdsv1

### PR DESCRIPTION
**Pull request by jeet4320**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
We would like to use imdsv2 use first and then imdsv1, so if users enforce imdsv2 on their EC2 instances, then DLC will automatically use imdsv2 and imdsv1 call will be avoided.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/deep_learning_container.py [hunk 1]
```diff
@@ -308,14 +308,17 @@ def main():
...
     logging.basicConfig(level=logging.ERROR)
     token = None
+    instance_id = None
+    region = None
+    token = get_imdsv2_token()
+    if token:
+        instance_id = _retrieve_instance_id(token)
 
-    instance_id = _retrieve_instance_id()
     if instance_id:
```
Comment (tejaschumbalkar) [inline src/deep_learning_container.py:317]: this condition can be completly avoided by adding `region = _retrieve_instance_region(token)` to the `if token:` condition

**Comment by tejaschumbalkar** [comment]
does our current EC2 instance to use imdsv2 in order to test this?

File: src/deep_learning_container.py (hunk 1 above)
Comment (kenny-ezirim) [reply src/deep_learning_container.py:317]: I think both functions _retrieve_instance_id() and _retrieve_instance_region should be able to handle the case where token has a value or is None|undefined. Then, there would be no need need for any of the conditions, and all you will have is the code below.
`token = get_imdsv2_token()`
`instance_id = _retrieve_instance_id(token)`
`region = _retrieve_instance_region(token)`

File: dlc_developer_config.toml [hunk 2]
```diff
@@ -28,7 +28,7 @@ benchmark_mode = false
 [build]
```
Comment (kenny-ezirim) [inline dlc_developer_config.toml:28]: I think you will need to use the updated dlc_developer_config.toml file to run the PR tests with the recent CI/CD changes.

**Pull request closed without merging**